compiler/qsc_qasm3/src/compile.rs

```diff
@@ -0,0 +1,4363 @@
+                if let Some(width) = width {
+                    if *width > 64 {
+                        Some(build_lit_bigint_expr(BigInt::from(0), span))
+                    } else {
+                        Some(build_lit_int_expr(0, span))
+                    }
+                } else {
+                    Some(build_lit_int_expr(0, span))
+                }
+            }
+            Type::Float(_, _) => Some(build_lit_double_expr(0.0, span)),
+            Type::Angle(_, _) => todo!("Angle default values"),
+            Type::Complex(_, _) => Some(build_lit_complex_expr(
+                crate::types::Complex::new(0.0, 0.0),
+                span,
+            )),
+            Type::Bool(_) => Some(build_lit_bool_expr(false, span)),
+            Type::Duration(_) => {
+                self.push_unsupported_error_message(
+                    "Duration type values are not supported.",
+                    node,
+                );
+                None
+            }
+            Type::Stretch(_) => {
+                self.push_unsupported_error_message("Stretch type values are not supported.", node);
+                None
+            }
+            Type::BitArray(dims, _) => match dims {
+                ArrayDims::D1(len) => Some(build_default_result_array_expr(*len, span)),
+                ArrayDims::D2(_, _) => {
+                    self.push_unsupported_error_message(
+                        "2-dim Bit Arrays without default values",
+                        node,
+                    );
+                    None
+                }
+                ArrayDims::D3(_, _, _) => {
+                    self.push_unsupported_error_message(
+                        "3-dim Bit Arrays without default values",
+                        node,
+                    );
+                    None
+                }
+            },
+            Type::QubitArray(_) => unimplemented!("QubitArray default values"),
+            Type::IntArray(_)
+            | Type::UIntArray(_)
+            | Type::FloatArray(_)
+            | Type::AngleArray(_)
+            | Type::ComplexArray(_)
+            | Type::BoolArray(_) => {
+                self.push_unsupported_error_message("Arrays without default values", node);
+                None
+            }
+            Type::DurationArray(_) => {
+                self.push_unsupported_error_message(
+                    "DurationArray type values are not supported.",
+                    node,
+                );
+                None
+            }
+            Type::Gate(_, _) => unimplemented!("Gate default values"),
+            Type::Range => unimplemented!("Range default values"),
+            Type::Set => unimplemented!("Set default values"),
+            Type::Void => unimplemented!("Void default values"),
+            Type::ToDo => unimplemented!("ToDo default values"),
+            Type::Undefined => unimplemented!("Undefined default values"),
```

Comment:
Are these reachable from Python? Will this crash the program if the user happens to try these from a Python Notebook?

Reply:
Those should be bugs and effectively unreachable. We can convert them to semantic errors, with the comment that it is unlikely that the cases can ever be hit.